larray/core.py

```diff
@@ -5069,6 +5165,76 @@ def get_labels(self_axis):
         else:
             return res
 
+    def align(self, other, join='outer', axes=None, fill_value=nan):
+        """Align two arrays on their axes with the specified join method.
+
+        In other words, it ensure all common axes are compatible.
+
+        Parameters
+        ----------
+        other : LArray-like
+        join : {'outer', 'inner', 'left', 'right'}, optional 
+            Defaults to 'outer'.
+        axes : AxisReference or sequence of them, optional
+            Axes to align. Need to be valid in both arrays. Defaults to None (all common axes).
+        fill_value : scalar, default np.NaN
+            Value to use for missing values. Defaults to NaN, but can be any
+            "compatible" value.
+
+        Returns
+        -------
+        (left, right) : (LArray, LArray)
+            Aligned objects
+
+        Examples
+        --------
+        >>> arr1 = ndtest((2, 3))
+        >>> arr1
+        a\\b | b0 | b1 | b2
+         a0 |  0 |  1 |  2
+         a1 |  3 |  4 |  5
+        >>> arr2 = ndtest((3, 2, 1))
+        >>> arr2
+         a | b\\c | c0
+        a0 |  b0 |  0
+        a0 |  b1 |  1
+        a1 |  b0 |  2
+        a1 |  b1 |  3
+        a2 |  b0 |  4
+        a2 |  b1 |  5
+        >>> aligned1, aligned2 = arr1.align(arr2)
+        >>> aligned1
+        a\\b |  b0 |  b1 |  b2
+         a0 | 0.0 | 1.0 | 2.0
+         a1 | 3.0 | 4.0 | 5.0
+         a2 | nan | nan | nan
+        >>> aligned2
+         a | b\\c |  c0
+        a0 |  b0 | 0.0
+        a0 |  b1 | 1.0
+        a0 |  b2 | nan
+        a1 |  b0 | 2.0
+        a1 |  b1 | 3.0
+        a1 |  b2 | nan
+        a2 |  b0 | 4.0
+        a2 |  b1 | 5.0
+        a2 |  b2 | nan
+        >>> aligned1 + aligned2
+         a | b\\c |  c0
+        a0 |  b0 | 0.0
+        a0 |  b1 | 2.0
+        a0 |  b2 | nan
+        a1 |  b0 | 5.0
+        a1 |  b1 | 7.0
+        a1 |  b2 | nan
+        a2 |  b0 | nan
+        a2 |  b1 | nan
+        a2 |  b2 | nan
+        """
+        other = aslarray(other)
+        left_axes, right_axes = self.axes.align(other.axes, join=join, axes=axes)
+        return self.reindex(left_axes, fill_value=fill_value), other.reindex(right_axes, fill_value=fill_value)
```

Comment:
If I understand well, AxisCollection.align returns the same AxisCollection `new_axes` twice. So why not returning only one `new_axes` collection and replace here left_axes, right_axes by common_axes?